sdk/notificationhubs/notification-hubs/src/utils/connectionStringUtils.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createSasTokenProvider, parseConnectionString, SasTokenProvider } from "@azure/core-amqp";
+
+/**
+ * The set of properties that comprise a Notification Hubs connection string.
+ */
+export interface NotificationHubsConnectionStringProperties {
+  /**
+   * The value for "Endpoint" in the connection string.
+   */
+  endpoint: string;
+  /**
+   * The value for "SharedAccessKey" in the connection string. This along with the "SharedAccessKeyName"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKey?: string;
+  /**
+   * The value for "SharedAccessKeyName" in the connection string. This along with the "SharedAccessKey"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKeyName?: string;
+}
+
+export function createTokenProviderFromConnectionString(connectionString: string): SasTokenProvider {
+  const parsed = parseNotificationHubsConnectionString(connectionString) as Required<
```

Comment:
what's the point in just casting the two optional properties as required? Either make them required on the interface or do a runtime check to see if they exist here

Reply:
Passing in the via the parsed connection now